plugins/costinsight/src/hooks/utility.js

```diff
@@ -0,0 +1,146 @@
+
+/*****************DO NOT EDIT CONTENT IN THIS FILE **************/
+
+import { Grid, Card, CardContent, makeStyles } from "@material-ui/core";
+
+/**
+ *  process response */
+
+export function convertSchema(schema, rows) {
+
+    var resultRows = []
+
+    function recurse(schemaCur, rowsCur, colName) {
```

Comment:
Should this be inside of the scope of another function?

Reply:
Yes, when enforce TS component.

Reply:
i can reduce one with **'withkeyfunction'**